flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/LookupJoinHintOptions.java

```diff
@@ -100,6 +100,12 @@ public class LookupJoinHintOptions {
                     .noDefaultValue()
                     .withDescription("Max attempt number of the 'fixed-delay' retry strategy.");
 
+    public static final ConfigOption<Boolean> SHUFFLE =
+            key("shuffle")
```

Comment:
Should it be "custom-shuffle" instead of "shuffle"? Shuffle should always be there, the difference is whether custom shuffle is used.

Reply:
This pr not only introduced custom shuffle, but also support hash shuffle for lookup join input stream. Without this shuffle hint, input stream is not distributed by join key.

Reply:
![image](https://github.com/user-attachments/assets/ce653352-2b41-4d71-93ed-32ede8b3a95e)
refer to FLIP

Reply:
refer to FLIP

![image](https://github.com/user-attachments/assets/83d17435-f1e7-45b1-bae0-ec4177471e3e)